rectools/metrics/scoring.py

```diff
@@ -196,4 +197,11 @@ def calc_metrics(  # noqa  # pylint: disable=too-many-branches,too-many-locals,t
 
     if len(results) < expected_results_len:
         warnings.warn("Custom metrics are not supported.")
+
+    # DQ
+    dq_metrics = select_by_type(metrics, DQMetric)
```

Comment:
I really don't like the mess we have here. Before we had at least all argument checks here and all metrics of the same kind accepted the same arguments, but now you have different metrics united as `dq` and checks are moved to its inner function...

Let's keep it for now, but we need to invent a better solution in future

Reply:
I agree. Maybe I just create another module?
`cross_dq.py` for `UsersNotCovered` and `dq.py` for other 2 metrics?
It'more structured in terms of our computation logic.

Reply:
hmm, let's discuss it separately as well

I'm afraid that if you add a new metrics that requires, say, `catalog` you will have to add one more file, and that scares me

Reply:
Decided to leave it as it is for now

Reply:
refactored the whole thing